### PR TITLE
remove ugly and unnecessary ignore result (_ = ...)

### PR DESCRIPTION
**Pull request by weissi**
Motivation:

_ = expression() is ugly and in many cases unimportant. In fact we train
ourselves to overread it which makes special cases where you'd actually
expect a result to be used just look normal.

Modifications:

remove _ = from a lot of places. In many cases we already had a better
(and sometimes cheaper way) to not return a value but in some cases
(mostly `remove` functions) I added `@discardableResult`

Result:

NIO source code looks nicer

### CODE REVIEW

File: Sources/NIOHTTP1/HTTPUpgradeHandler.swift [hunk 1]
```diff
@@ -236,8 +236,8 @@ public class HTTPServerUpgradeHandler: ChannelInboundHandler {
                     if bufferedMessages.count > 0 {
                         ctx.fireChannelReadComplete()
                     }
-                }.then {
-                    ctx.pipeline.remove(ctx: ctx)
+                }.whenComplete {
```
Comment (Lukasa) [inline Sources/NIOHTTP1/HTTPUpgradeHandler.swift]: This is not quite semantically equivalent, but I think it's more accurate anyway.

**Comment by Lukasa** [comment]
Re-tagged this as a server-minor as adding `@discardableResult` changes the API in a backward-compatible way.

**Comment by weissi** [comment]
@Lukasa how does it change the API? Anything that was valid before is still valid, no?

**Comment by Lukasa** [comment]
Ah, forgot that discarding a non-discardable result only produced a warning, not an error. You're right, this is fine: warnings don't affect semver.

**Comment by weissi** [comment]
@Lukasa also, it can also _remove_ a warning from existing code not add one

**Comment by Lukasa** [comment]
@weissi That's why it would have been a minor, not a major. A minor means that code written against the new code may not compile against the old, but code written against the old will compile against the new. If `@discardableResult` made *errors* go away, it'd be a minor.

**Pull request merged**